### PR TITLE
tests/release: add kafka-ingest-open-loop to release tests

### PR DESCRIPTION
**Pull request by ruchirK**
This commit adds two versions of the kafka-ingest-open loop test, one
with persistence to s3, and one without.

The tests are configured to run for 24 hours, with 1k inserts of 500 byte values
per second. Previously, when I ran 100k records/second for 100 seconds Materialize
used ~2.5 GB of RAM without persistence, and approximately 14 GB of RAM with
persistence. This test will send approximately 9x as much data, over a ~900x larger
time interval, for a total of ~90 million inserts (45 GB total data).

### Checklist

- [ ] This PR has adequate test coverage / QA involvement has been duly considered.
- [ ] This PR adds a release note for any [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/user/content/release-notes.md#what-changes-require-a-release-note).

### CODE REVIEW

File: misc/scratch/release.json [hunk 1]
```diff
@@ -1,31 +1,29 @@
...
-        "test": "chbench",
+        "test": "kafka-ingest-open-loop",
         "environment": "scratch"
     }
 }
 
 {
-    "name": "kinesis",
-    "launch_script": "MZ_WORKERS=4 bin/mzcompose --preserve-ports --find perf-kinesis run load-test",
+    "name": "kafka-ingest-open-loop-s3",
```
Comment (danhhz) [inline misc/scratch/release.json:17]: I think to make it clear that these are persist vs nopersist (as opposed to something like persist-file vs persist-s3), I'd name this one "kafka-ingest-open-loop-persist". Given that S3 will be our production storage, we don't need to mention it

**Review by danhhz — COMMENTED** [review]
looks like you replaced the existing tests instead of adding new ones :)

**Comment by ruchirK** [comment]
ah shoot sorry that was my local deployment testing didn't mean to push that lol

File: misc/scratch/release.json (hunk 1 above)
Comment (ruchirK) [reply misc/scratch/release.json]: done

**Comment by ruchirK** [comment]
Fixed! Am running these now to make sure everything works as expected

File: misc/scratch/release.json [hunk 2]
```diff
@@ -29,3 +29,33 @@
...
+        "lt_name": "release-kafka-ingest-open-loop",
+        "purpose": "load_test",
+        "test": "kafka-ingest-open-loop",
+        "environment": "scratch"
+    }
+}
+
+{
+    "name": "kafka-ingest-open-loop-persist",
+    "launch_script": "MZ_WORKERS=4 bin/mzcompose --preserve-ports --find kafka-ingest-open-loop run default --num-seconds=86400 --records-per-second=1000 --enable-persistence --s3-storage=release-$(echo $RANDOM)",
```
Comment (danhhz) [inline misc/scratch/release.json:50]: TIL about `echo $RANDOM`! I always use date for this but getting the formatting nice and then working across both linux and macos versions of date is fiddly

**Review by danhhz — APPROVED** [review]

**Comment by ruchirK** [comment]
hmmm I'm running into a weird thing where when I run everything here without `--preserve-ports` it works fine, but with `--preserve-ports` mzcompose fails to verify that zookeeper started. still investigating but will mark as draft to make sure it doesn't get accidentally merged

**Comment by ruchirK** [comment]
I can make everything work with the following diff

```
ubuntu@ip-10-1-25-233:~/materialize$ git diff
diff --git a/test/kafka-ingest-open-loop/mzcompose.py b/test/kafka-ingest-open-loop/mzcompose.py
index 831d85fba..e11304722 100644
--- a/test/kafka-ingest-open-loop/mzcompose.py
+++ b/test/kafka-ingest-open-loop/mzcompose.py
@@ -166,7 +166,8 @@ def workflow_default(c: Composition, parser: WorkflowArgumentParser) -> None:
     ]

     with c.override(*override):
-        c.start_and_wait_for_tcp(services=prerequisites)
+        for service in prerequisites:
+            c.up(service)

         c.up("materialized")
         c.wait_for_materialized("materialized")
```

but unclear if there's a better fix. On the plus side - we've got this benchmark with s3 all hooked up to grafana finally :)

**Comment by ruchirK** [comment]
this is all working as expected now! I'm going to let the test run for 24 hours just to make sure we don't completely OOM the machine when running with persist (I think we should be fine but doesn't hurt to double check). TFTR!

**Comment by ruchirK** [comment]
Materialize survived 21 hours (and counting) of this test running with s3 so I'm going to call this a success. TFTR!

**Pull request merged**